packages/data-stores/src/domain-suggestions/resolvers.ts

```diff
@@ -54,34 +54,33 @@ export function* __internalGetDomainSuggestions( queryObject: DomainSuggestionQu
 
 	yield fetchDomainSuggestions();
 
-	let suggestions;
 	try {
-		suggestions = yield wpcomRequest( {
+		const suggestions: string | DomainSuggestion[] = yield wpcomRequest( {
 			apiVersion: '1.1',
 			path: '/domains/suggestions',
 			query: stringify( queryObject ),
 		} );
+
+		if ( typeof suggestions === 'string' || ! suggestions ) {
```

Comment:
Since `suggestions` is used as an array below, what if we did this instead: (It should also cover the falsy check)

```suggestion
		if ( ! Array.isArray( suggestions ) ) {
```

Reply:
Fixed